### PR TITLE
Add /v1/users/register and database supports

### LINKED ISSUE #1: U1: Visitor can register using email and password
```javascript
Endpoint: 
POST /v1/users/register
Request:
{
  userName: String,
  displayName: String,
  email: String,
  password: String,
}
Response:
{
  status: String,
  payload: any,
}
Success:
{status: 'SUCCESS'}
Error:
{status: 'ERROR', payload: '<Error Message>'}
```

### PR DESCRIPTION
**Pull request by HinataIzayoi**
Process POST request from /v1/users/register.

Return API error if User with the same userName or email already exists.
Create new User in database if they does not exist.

Resolve #1

### CODE REVIEW

File: controllers/userController.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const User = require('../models/User');
+const {apiError, apiSuccess, genPasswordSalt, calcPasswordHash} = require('./utils');
+
+exports.register = async function(params) {
+  const existingUser = await User.findOne({
```
Comment (Kailang) [inline controllers/userController.js]: Do not use findOne for checking existence. Use count() instead.

File: controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const User = require('../models/User');
+const {apiError, apiSuccess, genPasswordSalt, calcPasswordHash} = require('./utils');
```
[the remark this answers is not in the document]
Comment (HinataIzayoi) [reply controllers/userController.js]: Thanks.

File: .eslintrc.json [hunk 3]
```diff
@@ -1,19 +1,19 @@
 {
-   "extends": ["eslint:recommended", "google"],
```
[the remark this answers is not in the document]
Comment (HinataIzayoi) [reply .eslintrc.json]: Revert the indent changes.

Created the issue #3 .
Will do the reformat in another PR.

File: controllers/userController.js (hunk 1 above)
Comment (HinataIzayoi) [reply controllers/userController.js]: Done.

**Pull request merged**